escrow/contracts/Escrow.cdc

```diff
@@ -141,24 +135,27 @@ pub contract Escrow {
     }
 
     // An interface containing the Collection function that gets leaderboards by name.
-    pub resource interface ICollectionPublic {
-        pub fun getLeaderboardInfo(name: String): LeaderboardInfo?
-        pub fun addEntryToLeaderboard(nft: @NonFungibleToken.NFT, leaderboardName: String, ownerAddress: Address, metadata: {String: AnyStruct})
+    access(all) resource interface ICollectionPublic {
+        access(all) fun getLeaderboardInfo(name: String): LeaderboardInfo?
+        access(all) fun addEntryToLeaderboard(nft: @{NonFungibleToken.NFT}, leaderboardName: String, ownerAddress: Address, metadata: {String: AnyStruct})
```

Comment:
interesting.. why `addEntryToLeaderboard` doesn't require any access control?

Reply:
The intent is to expose `addEntryToLeaderboard` publicly, though it can only be called by providing a NFT of type as specified in the leaderboard. cc @Deewai @chumeston